FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.JobManagement/AzureStorageJobQueueClient.cs

```diff
@@ -0,0 +1,717 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using Azure;
+using Azure.Data.Tables;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Exceptions;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Extensions;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models.AzureStorage;
+using Microsoft.Health.JobManagement;
+
+namespace Microsoft.Health.Fhir.Synapse.JobManagement
+{
+    // The maximum size of a single entity, including all property values is 1 MiB,
+    // see https://docs.microsoft.com/en-us/azure/storage/tables/scalability-targets#scale-targets-for-table-storage.
+    // The definition and result are serialized string of objects,
+    // we should be careful that should not contain large fields when define the definition and result class.
+    public class AzureStorageJobQueueClient<TJobInfo> : IQueueClient
+        where TJobInfo : AzureStorageJobInfo, new()
+    {
+        private readonly IAzureStorageClient _azureStorageClient;
+        private readonly TableClient _azureJobInfoTableClient;
+        private readonly QueueClient _azureJobMessageQueueClient;
+
+        private readonly ILogger<AzureStorageJobQueueClient<TJobInfo>> _logger;
+
+        private const int DefaultVisibilityTimeoutInSeconds = 30;
+
+        private const short MaxThreadsCountForGettingJob = 5;
+
+        public AzureStorageJobQueueClient(
```

Comment:
I believe we do not need IStorage

```
public AzureStorageJobQueueClient(
IAzureStorageClientFactory azureStorageClientFactory,
ILogger<AzureStorageJobQueueClient<TJobInfo>> logger)
```

IAzureStorageClientFactory
- CreateTableClient()
- CreateQueueClient()

Reply:
`IStorage` is removed now.